src/megameklab/com/ui/battleArmor/BABuildView.java

```diff
@@ -275,7 +279,7 @@ public void actionPerformed(ActionEvent e) {
                     item.addActionListener(new ActionListener() {
                         public void actionPerformed(ActionEvent e) {
                             eq.setLocation(BattleArmor.LOC_TROOPER_1);
-                            ((BuildTab) getParent().getParent()).refreshAll();
+                            ((BABuildTab) getParent().getParent()).refreshAll();
```

Comment:
Any chance the two getParent methods could be overridden to instead just directly return the build tab?

Reply:
These are Swing internal Component methods I'd rather not override. These calls don't look like a good idea overall but I havent looked at them before and think it's beyond this particular PR to change.